sdk/eventhub/event-hubs/test/eventProcessor.spec.ts

```diff
@@ -1092,65 +1093,93 @@ describe("Event Processor", function(): void {
         await producer.close();
       }
 
+      const processor1LoadBalancingInterval = {
+        loopIntervalInMs: 1000
+      };
+
+      // working around a potential deadlock - this allows `processor-2` to more
+      // aggressively pursue getting its required partitions and avoid being in
+      // lockstep with `processor-1`
+      const processor2LoadBalancingInterval = {
+        loopIntervalInMs: processor1LoadBalancingInterval.loopIntervalInMs/2
+      };
+
       processorByName[`processor-1`] = new EventProcessor(
         EventHubClient.defaultConsumerGroupName,
         client,
         new FooPartitionProcessor(),
         checkpointStore,
-        { ...defaultOptions, startPosition: earliestEventPosition }
+        { ...defaultOptions, startPosition: earliestEventPosition, ...processor1LoadBalancingInterval }
       );
 
       processorByName[`processor-1`].start();
 
-      while (partitionOwnershipArr.size !== partitionIds.length) {
-        loggerForTest("Waiting for partition ownership");
-        await delay(5000);
-      }
+      await loopUntil({
+        name: "All partitions are owned",
+        maxTimes: 60,
+        timeBetweenRunsMs: 1000,
+        until: async () => partitionOwnershipArr.size === partitionIds.length,
+        errorMessageFn: () => `${partitionOwnershipArr.size}/${partitionIds.length}`
+      });
 
       processorByName[`processor-2`] = new EventProcessor(
         EventHubClient.defaultConsumerGroupName,
         client,
         new FooPartitionProcessor(),
         checkpointStore,
-        { ...defaultOptions, startPosition: earliestEventPosition }
+        { ...defaultOptions, startPosition: earliestEventPosition, ...processor2LoadBalancingInterval }
       );
 
       partitionOwnershipArr.size.should.equal(partitionIds.length);
       processorByName[`processor-2`].start();
 
-      loggerForTest(`Just before the big arbitrary delay`);
-      await delay(12000);
-      loggerForTest(`Just after the big arbitrary delay`);
+      await loopUntil({
+        name: "Processors have reached equilibium",
```

Comment:
```suggestion
        name: "Processors have reached equilibrium",
```

Reply:
Just changed it to use the term "balanced" since that's our terminology.